app/models/order.rb

```diff
@@ -0,0 +1,27 @@
+class Order < ApplicationRecord
+  has_many :order_items
+  accepts_nested_attributes_for :order_items
+  belongs_to :user
+  validates :total_amount, presence: true, numericality: { greater_than: 0 }
```

Comment:
Remember to validate each individual item also (cart rows with negative amounts is actually possible)